tnt/v2/tnt_v2.2.0.yaml

```diff
@@ -0,0 +1,1970 @@
+          example: APZU4812090
+          maxLength: 15
+    Subscription:
+      title: Subscriptionn
+      x-stoplight:
+        id: 9k2fwmxqe1ho6
+      allOf:
+        - type: object
+          properties:
+            subscriptionID:
+              type: string
+              description: The carrier issues a unique ID to the shipper or consignee for that subscription.
+              example: 8fbdc2d8-57c8-48b9-a04b-18fd8ec1d809
+              format: uuid
+        - $ref: '#/components/schemas/SubscriptionBody'
+    ShipmentEventTypeCode:
+      title: ShipmentEventTypeCode
+      x-stoplight:
+        id: 6f9r35u4o5rrv
+      type: string
+      description: |
+        The status of the document in the process. Possible values are
+        - RECE (Received)
+        - DRFT (Drafted)
+        - PENA (Pending Approval)
+        - PENU (Pending Update)
+        - REJE (Rejected)
+        - APPR (Approved)
+        - ISSU (Issued)
+        - SURR (Surrendered)
+        - SUBM (Submitted)
+        - VOID (Void)
+        - CONF (Confirmed)
+        - REQS (Requested)
+        - CMPL (Completed)
+        - HOLD (On Hold)
+        - RELS (Released)
+
+        More details can be found on <a href="https://github.com/dcsaorg/DCSA-Information-Model/blob/master/datamodel/referencedata.d/shipmenteventtypecodes.csv">GitHub</a>
+      enum:
+        - RECE
+        - DRFT
+        - PENA
+        - PENU
+        - REJE
+        - APPR
+        - ISSU
+        - SURR
+        - SUBM
+        - VOID
+        - CONF
+        - REQS
+        - CMPL
+        - HOLD
+        - RELS
+      example: DRFT
+    DocumentTypeCode:
+      title: DocumentTypeCode
+      x-stoplight:
+        id: n0b50jkkgv9wq
+      type: string
+      description: |-
+        The code to identify the type of information documentID points to. Can be one of the following values 
+        - CBR (Carrier Booking Request Reference) 
+        - BKG (Booking) - SHI (Shipping Instruction) 
+        - SRM (Shipment Release Message) 
+        - TRD (Transport Document) 
+        - ARN (Arrival Notice) 
+        - VGM (Verified Gross Mass) 
+        - CAS (Cargo Survey) 
+        - CUS (Customs Inspection) 
+        - DGD (Dangerous Goods Declaration) 
+        - OOG (Out of Gauge)
+
+        More details can be found on <a href="https://github.com/dcsaorg/DCSA-Information-Model/blob/master/datamodel/referencedata.d/documenttypecodes.csv">GitHub</a>
+      example: SHI
+      enum:
+        - CBR
+        - BKG
+        - SHI
+        - SRM
+        - TRD
+        - ARN
+        - VGM
+        - CAS
+        - CUS
+        - DGD
+        - OOG
+    SunscriptionBodyWithSecret:
+      title: SunscriptionBodyWithSecret
+      x-stoplight:
+        id: bcisr6k4u6kiv
+      allOf:
+        - $ref: '#/components/schemas/SubscriptionBody'
+        - type: object
+          required:
+            - secret
```

Comment:
I am not sure if this produces the correct result. The idea is that the `secret` property needs to be a required field on the `SubscriptionBodyWithSecret` object. If it does so - then disregard this comment

Reply:
It least it is not defined in this way on the T&T 2.2 spec on SwaggerHub